### PR TITLE
JIT IR tidyups

### PR DESCRIPTION
**Pull request by ltratt**
This continues the general tidying up of the JIT IR. Each commit hopefully stands alone.

### CODE REVIEW

File: ykrt/src/compile/jitc_yk/jit_ir.rs [hunk 1]
```diff
@@ -211,13 +211,14 @@ impl Module {
     }
 
     /// Push an instruction to the end of the [Module].
-    ///
-    /// # Panics
-    ///
-    /// If `instr` would overflow the index type.
-    pub(crate) fn push(&mut self, inst: Inst) {
-        assert!(InstIdx::new(self.insts.len()).is_ok());
```
Comment (vext01) [inline ykrt/src/compile/jitc_yk/jit_ir.rs:219]: The `assert!` here makes `push()` crash if the index overflows, so the commit message isn't accurate, right? ("Although previously the code didn't actually panic")

I do actually prefer rippling the error up though, so the change is good, but the commit message isn't.

File: ykrt/src/compile/jitc_yk/jit_ir.rs [hunk 2]
```diff
@@ -211,13 +211,14 @@ impl Module {
...
-        self.insts.push(inst);
+    pub(crate) fn push(&mut self, inst: Inst) -> Result<InstIdx, CompilationError> {
+        match InstIdx::new(self.insts.len()) {
```
Comment (vext01) [inline ykrt/src/compile/jitc_yk/jit_ir.rs:215]: Can this be:
```
InstIdx::new(self.insts.len()).and_then(|i| {
    self.insts.push(inst);
    Ok(i)
})
```

**Comment by vext01** [comment]
Just a couple of minor bits.

File: ykrt/src/compile/jitc_yk/jit_ir.rs (hunk 1 above)
Comment (ltratt) [reply ykrt/src/compile/jitc_yk/jit_ir.rs:219]: Oops, yes, you're right about the `panic`!

File: ykrt/src/compile/jitc_yk/jit_ir.rs (hunk 2 above)
Comment (ltratt) [reply ykrt/src/compile/jitc_yk/jit_ir.rs:215]: Yeah, I experimented with that, but I wasn't sure it was clearer with the mutation? I'm easy though.

Comment (vext01) [reply ykrt/src/compile/jitc_yk/jit_ir.rs:215]: Neither way is fantastic. At least this is shorter? Won't die on that hill.

File: ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs [hunk 3]
```diff
@@ -300,7 +300,7 @@ impl<'a> X64CodeGen<'a> {
     fn load_const(&mut self, reg: Rq, cidx: jit_ir::ConstIdx) {
         let cst = self.m.const_(cidx);
         let mut bytes = cst.bytes().as_slice();
-        let size = cst.ty_idx().type_(self.m).byte_size().unwrap();
+        let size = self.m.type_(cst.ty_idx()).byte_size().unwrap();
```
[the remark this answers is not in the document]
Comment (vext01) [reply ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs:303]: OK

File: ykrt/src/compile/jitc_yk/jit_ir.rs (hunk 2 above)
Comment (ltratt) [reply ykrt/src/compile/jitc_yk/jit_ir.rs:215]: I'm inclined to leave it as-is TBH.

**Comment by ltratt** [comment]
OK to force push an update with a fixed commit message?

**Comment by vext01** [comment]
Go ahead.

**Comment by ltratt** [comment]
Commit message updated.

**Pull request merged**